src/main/java/seedu/address/model/DeckModelManager.java

```diff
@@ -18,28 +18,27 @@
 /**
  * Represents the in-memory model of the deck data.
  */
-public class CardModelManager implements CardModel {
-    private static final Logger logger = LogsCenter.getLogger(CardModelManager.class);
-
-    private final PowerDeck deck;
+public class DeckModelManager implements DeckModel {
+    private static final Logger logger = LogsCenter.getLogger(DeckModelManager.class);
+    private PowerDeck selectedDeck = null;
+    private final ArrayList<PowerDeck> powerDecks = new ArrayList<>();
     private final UserPrefs userPrefs;
-    private final FilteredList<PowerCard> filteredCards;
+    //  private final FilteredList<PowerCard> filteredCards;
 
     /**
      * Initializes a CardModelManager with the given deck and userPrefs.
      */
-    public CardModelManager(ReadOnlyPowerDeck powerDeck, ReadOnlyUserPrefs userPrefs) {
-        requireAllNonNull(powerDeck, userPrefs);
+    public DeckModelManager(ReadOnlyUserPrefs userPrefs) {
```

Comment:
I think the reason why the constructor takes in a ReadOnlyPowerDeck is so that it can load the save file in Storage. If we remove this parameter there is currently no way to load the save file...

Reply:
![image](https://user-images.githubusercontent.com/80334013/223472655-2005fe94-eac6-4beb-abe1-af7f1d4c78e5.png)

This method in MainApp class is responsible for loading the addressbook from storage using the constructor.